src/Bridge/Graphql/Type/SchemaBuilder.php

```diff
@@ -0,0 +1,298 @@
+                        'description' => 'Returns the first n elements from the list.'
+                    ],
+                ],
+                'resolve' => function ($root, $args, $context, ResolveInfo $resolveInfo) use ($resource, $operationName) {
+                    // @TODO Pagination
+                    $collection = $this->collectionDataProvider->getCollection($resource, $operationName);
+                    return $this->normalizer->normalize($collection, null, ['graphql_selected_fields' => $resolveInfo->getFieldSelection()]);
+                },
+            ];
+        }
+
+        return $queryFields;
+    }
+
+    /**
+     * Gets the type of the given resource.
+     *
+     * @param string           $resource
+     * @param ResourceMetadata $resourceMetadata
+     *
+     * @return ObjectType
+     */
+    private function getResourceType(string $resource, ResourceMetadata $resourceMetadata): ObjectType
+    {
+        $shortName = $resourceMetadata->getShortName();
+
+        if (!isset($this->resourceTypesCache[$shortName])) {
+            $this->resourceTypesCache[$shortName] = new ObjectType([
+                'name' => $shortName,
+                'description' => $resourceMetadata->getDescription(),
+                'fields' => function () use ($resource) {
+                    return $this->getResourceTypeFields($resource);
+                },
+            ]);
+        }
+
+        return $this->resourceTypesCache[$shortName];
+    }
+
+    /**
+     * Gets the fields of the type of the given resource.
+     *
+     * @param string $resource
+     *
+     * @return array
+     */
+    private function getResourceTypeFields(string $resource): array
+    {
+        $fields = [];
+
+        foreach ($this->propertyNameCollectionFactory->create($resource) as $property) {
+            $propertyMetadata = $this->propertyMetadataFactory->create($resource, $property);
+
+            if (null !== $propertyMetadata->getType()) {
+                try {
+                    $fields[$property] = [
+                        'type' => $this->convertType($propertyMetadata->getType()),
+                        'description' => $propertyMetadata->getDescription(),
+                    ];
+                } catch (InvalidTypeException $e) {
+                    continue;
+                }
+            }
+        }
+
+        return $fields;
+    }
+
+    /**
+     * Gets the field arguments of the identifier of a given resource.
+     *
+     * @param string $resource
+     *
+     * @return array
+     */
+    private function getResourceIdentifierFieldArguments(string $resource): array
+    {
+        foreach ($this->propertyNameCollectionFactory->create($resource) as $property) {
+            $propertyMetadata = $this->propertyMetadataFactory->create($resource, $property);
+
+            if (!$propertyMetadata->isIdentifier()) {
+                continue;
+            }
+
+            // @TODO Manage multiple identifiers?
```

Comment:
Maybe can you use just reuse this method: https://github.com/api-platform/core/blob/master/src/Bridge/Doctrine/Orm/Util/IdentifierManagerTrait.php#L40